src/main/java/org/plp/isa/AsmAssemblerDirectiveProcessor.java

```diff
@@ -0,0 +1,20 @@
+package org.plp.isa;
+
+/**
+ * This processes a type of assembler directives based on the {@link AsmArgument}s passed to it
+ */
+public interface AsmAssemblerDirectiveProcessor {
+    /**
+     * This processes the assembler directive based on the arguments passed and returns the
+     * replacement placeholder for further assembling of the assembly program
+     * @param asmArguments list of arguments in the order needed for processing directive
+     * @return replacement placeholder instruction to be used during assembly
+     */
+    String perform(AsmArgument[] asmArguments);
```

Comment:
Multiple asmArguments but only one String? Should it be one replacement placeholder for one argument?

Reply:
Example PLP has assembler directives like  `org 0x10000000` or `.asciiw "example string"` so each of these directives have single argument but at the end this will get transformed into something like `ASM__ORG__123456778` or `ASM__WORD__3435232` something like that. Even if there are multiple it will be `\n` separated as this is in the preprocessing stage and we need to maintain line number parity with the actual program.

Reply:
So for AsmAssemblerDirectiveProcessor, the perform operation takes in only one argument all the time?

In this case,
```
String perform(AsmArgument asmArgument);
```
will be better?

In AsmPseudoInstructionAssembler will return the `\n` separated subinstructions.

Reply:
An assembler directive can have multiple arguments. 
- Example of single argument assemblerDirective `.org 0x10000000` 
- Example of multiple argument assembler Directive `.equ mylocation 0x200000`

So PlpOrgAssemblerDirectiveProcessor needs to handle single argument `0x100000` where as PlpEquAssemblerDirectiveProcessor needs to handle 2 arguments `mylocation` and `0x200000`. 

So interface is saying an assembler directive can have multiple arguments but individual implementation will decide the actual number of arguments needed for their directive.

Reply:
Sounds good, so the format of this string on multiple arguments decided by the actual implementation. We only guarantee the single line (line number should be the same) during preprocessing.

I think we can add this information to the comment.

Reply:
Yes adding it to the comments

Reply:
Added to comments